bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -149,6 +158,69 @@ protected void deactivate() {
         newCategoryUrlFormat = null;
     }
 
+    private ProductUrlFormat getProductFormatUrlFormat(SlingHttpServletRequest request, Page page) {
+        if (request == null && page == null) {
+            throw new IllegalArgumentException("The request and the page parameters cannot both be null");
+        }
+
+        ProductUrlFormat productUrlFormat = newProductUrlFormat;
+        Resource resource = request != null ? request.getResource() : page.getContentResource();
+
+        ComponentsConfiguration properties = resource
+            .adaptTo(ComponentsConfiguration.class);
+
+        if (properties != null) {
+            String formatPattern = properties.get(PN_PRODUCT_PAGE_URL_FORMAT, String.class);
+
+            if (formatPattern != null) {
+                if (DEFAULT_PRODUCT_URL_FORMATS.containsKey(formatPattern)) {
+                    productUrlFormat = DEFAULT_PRODUCT_URL_FORMATS.get(formatPattern);
+                } else if (productPageUrlFormat != null && !productPageUrlFormat.isEmpty()) {
+                    UrlFormat customUrlFormat = productPageUrlFormat.stream()
```

Comment:
You should also check for the `newProductUrlFormat` containing the custom format.